pkg/controller/utils/auth.go

```diff
@@ -69,16 +70,33 @@ func GetKeyValidatorConfig(ctx context.Context, cli client.Client, authenticatio
 				return nil, err
 			}
 		} else {
-			keyValidatorConfig = render.NewDexKeyValidatorConfig(authenticationCR, idpSecret, clusterDomain)
+			keyValidatorConfig = render.NewDexKeyValidatorConfig(authenticationCR, idpSecret, secretProviderClass, clusterDomain)
 		}
 	}
 
 	return keyValidatorConfig, nil
 }
 
-// GetIDPSecret retrieves the Secret containing sensitive information for the configuration IdP specified in the given
-// operatorv1.Authentication CR.
-func GetIDPSecret(ctx context.Context, client client.Client, authentication *operatorv1.Authentication) (*corev1.Secret, error) {
+func GetSecrets(ctx context.Context, client client.Client, authentication *operatorv1.Authentication) (*csisecret.SecretProviderClass, *corev1.Secret, error) {
+	secretProviderClass, err := GetSecretProviderClass(ctx, client, authentication)
+	if err == nil {
+		return secretProviderClass, nil, nil
+	}
+
+	idpSecret, err := GetIDPSecret(ctx, client, authentication)
+	return nil, idpSecret, err
+}
+
+func GetSecretProviderClass(ctx context.Context, client client.Client, authentication *operatorv1.Authentication) (*csisecret.SecretProviderClass, error) {
+	var secretName, _ = GetSecretNameAndRequiredFields(authentication)
+	secretProviderClass := &csisecret.SecretProviderClass{}
+	if err := client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: common.OperatorNamespace()}, secretProviderClass); err != nil {
+		return nil, fmt.Errorf("missing secret provider class %s/%s: %w", common.OperatorNamespace(), secretName, err)
```

Comment:
I think we should swallow errors when the secret provider class isn't there. If an error is returned, then GetSecrets(...) should return it to the caller.